src/FakeItEasy/Guard.cs

```diff
@@ -15,7 +16,7 @@ internal static class Guard
         /// <param name="argumentName">Name of the argument.</param>
         /// <exception cref="ArgumentNullException">The specified argument was null.</exception>
         [DebuggerStepThrough]
-        public static void AgainstNull([ValidatedNotNull]object argument, string argumentName)
+        public static void AgainstNull([ValidatedNotNull, NotNull] object? argument, string argumentName)

```

Comment:
It's a bit counter-intuitive IMO, but `[NotNull]` is a post-condition: it doesn't prevent passing null for `argument`, but means that if the method returns, then `argument` is not null.
I'm not sure what `[ValidatedNotNull]` is; looks like it comes from Code Contracts, which I'm not sure is still supported...

Reply:
> It's a bit counter-intuitive IMO, but [NotNull] is a post-condition: it doesn't prevent passing null for argument, but means that if the method returns, then argument is not null.

:+1: